remappers/k8smetrics/k8smetrics.go

```diff
@@ -0,0 +1,88 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package kubernetesmetrics
+
+import (
+	"path"
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+type remapFunc func(metrics pmetric.MetricSlice, out pmetric.MetricSlice, resource pcommon.Resource) error
+
+// Remapper maps the OTel hostmetrics to Elastic system metrics. These remapped
+// metrics power the curated Kibana dashboards. Each datapoint translated using
+// the remapper has the `event.processor` attribute set to `kubernetes`.
+type Remapper struct {
```

Comment:
it inlcudes the `cfg    config` variable that is specific to each package, eg.

 type config struct {
	KubernetesIntegrationDataset bool
}

I would say to leave it for now as it would require other changes. WDYT?